installinator/src/dispatch.rs

```diff
@@ -215,23 +245,45 @@ impl InstallOpts {
             WriteDestination::in_directory(destination)?
         };
 
-        let mut writer = ArtifactWriter::new(
-            &host_phase_2_id,
-            &host_phase_2_artifact.artifact,
-            &control_plane_id,
-            &control_plane_artifact.artifact,
-            destination,
-        );
-
-        writer.write(&log, &event_sender).await;
-
-        // TODO: verify artifact was correctly written out to disk.
-
-        // Drop the event sender: this signals completion.
-        _ = event_sender
-            .send(ReportEvent::Completion(CompletionEventKind::Completed))
-            .await;
-        std::mem::drop(event_sender);
+        // XXX: look at breaking this up into separate steps. This is tricky
+        // because of the combined logic around handling retries.
+        engine
+            .new_step(
+                InstallinatorComponent::Both,
+                InstallinatorStepId::Write,
+                "Writing host and control plane artifacts",
+                |cx| async move {
+                    let host_phase_2_artifact =
+                        host_phase_2_artifact.into_value(cx.token()).await;
+                    let control_plane_artifact =
+                        control_plane_artifact.into_value(cx.token()).await;
+
+                    let mut writer = ArtifactWriter::new(
+                        &host_2_phase_id_2,
+                        &host_phase_2_artifact.artifact,
+                        &control_plane_id_2,
+                        &control_plane_artifact.artifact,
+                        destination,
+                    );
+
+                    let slots_written = writer.write(&cx, &log).await;
+
+                    // TODO: warning message in case one of the slots wasn't
```

Comment:
Were you planning to add these in this PR or a follow up? Just flagging in case you forgot.

Reply:
Thanks for catching this! I did meant to add these in this PR, will fix.

Reply:
Done.